### DISCUSSION TITLE
Help with simple script that read from microphone

### DISCUSSION THREAD

**Discussion by guillermoAlv** [deepgram/community#72] [General help]
Sorry guys, but can someone help me creating a simple script that read from my microphone and send in real time the audio to Deepgram to transcribe it. I know that there is several example but all of them use Flask, Django or some others web frameworks, I've triying it for the last week but not lucky by now. If you need my script i can send you. BTW I'm working with the python SDK.

In the other hand what audio encode  needs Deepgram, or what format it interpret? just in order to debug my program.
Thanks a lot!

**Comment by geekchick** [comment]
Hi @guillermoAlv !

You can try something like the following in this blog post tutorial which uses PyAudio to get the audio from the computer's mic: https://blog.deepgram.com/how-to-transcribe-only-what-you-need-with-python-listening-before-connected/

Also, are you using a Windows or Mac machine? I'm asking because if you go the PyAudio route, it's a bit more involved on an M1 machine to get it installed.

**Reply by guillermoAlv** [reply]
Hi, thanks a lot for your response.
I'm using a M2 machine, I tried with Pyaudio but I was not able to make it works son finally I decided to use Sounddevice library. Do you have any example with Sounddevice library? I've tried using the script that is included in the Readme file in the python sdk repo (replacing the  radio stream with a RawInputStream from Sounddevice), but still no success.

**Reply by BekahHW** [reply]
@guillermoAlv sorry for the delay in response. This got a bit lost through the holidays. [This post](https://blog.deepgram.com/best-python-audio-manipulation-tools/) talks about using Sounddevice. Let me know if that's helpful or if you found a different solution.

**Reply by guillermoAlv** [reply]
Hi @BekahHW , thanks a lot for tour help.
The post is useful but does not solve my problem a need transcribe audio in real time, but the post Talk about recording audio. Do you have any example about transcribe audio in real time with sounddevice that not implies to use web frameworks?
Thanks again for tour time and help

**Reply by BekahHW** [reply]
I don't have another post, but can you describe what you've tried? Is the issue implementing Deepgram with Sounddevice and streaming or with how to use Sounddevice with streaming?
